vm/main.go

```diff
@@ -49,12 +49,15 @@ func main() {
 	}
 	router.Listener = ln
 
-	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	cliFactory := func() (*client.Client, error) {
+
+		return client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	}
```

Comment:
Seeing that we have to create a factory to ensure the client uses the right socket, I wish the `docker/docker/client` package did it by itself... It may be a common issue that if the client instance is kept in memory it may break because the socket has changed on the filesystem. :/

Reply:
nope 🤷